internal/iam/resource.go

```diff
@@ -49,28 +48,29 @@ type ResourceWithScope interface {
 
 // LookupScope looks up the resource's  scope
 func LookupScope(ctx context.Context, reader db.Reader, resource ResourceWithScope) (*Scope, error) {
+	const op = "iam.LookupScope"
 	if reader == nil {
-		return nil, stderrors.New("error reader is nil for LookupScope")
+		return nil, errors.New(errors.InvalidParameter, op, "missing reader")
 	}
 	if resource == nil {
-		return nil, stderrors.New("error resource is nil for LookupScope")
+		return nil, errors.New(errors.InvalidParameter, op, "missing resource")
 	}
 	if resource.GetPublicId() == "" {
-		return nil, fmt.Errorf("LookupScope: scope id is unset %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, op, "missing public id")
```

Comment:
The old error message said scope id is unset but this was after doing a `resource.GetPublicId() == ""` so I updated this to `missing public id` to align with other errors messages, let me know if I should keep it as `missing scope id`